bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/internal/hookregistry/HookRegistry.java

```diff
@@ -113,6 +114,7 @@ public void initialize() {
 			addClassLoaderHook(new DevClassLoadingHook(container.getConfiguration()));
 			addClassLoaderHook(new EclipseLazyStarter(container));
 			addClassLoaderHook(new WeavingHookConfigurator(container));
+			addClassLoaderHook(new ServiceLoaderMediatorHook());
```

Comment:
I considered this as well, mainly for the point you mentioned below that it would allow later Java versions than 8 (9 would already be a big win).
But the big drawback of this approach is that one has to list the fragment as framework-extension in a system-property of the runtime. For example in a launch config:
`-Dosgi.framework.extensions=reference:file:${project_loc:/org.eclipse.equinox.spi}`, in a product definition this would usually be `-Dosgi.framework.extensions=reference:file:org.eclipse.equinox.spi`, assuming the spi and o.e.osgi jar are colocated.
Or is there another way to load a framework-extension with the framework's classloader so that the `hookconfigurators.properties` are found?
Because actually another goal of this was to avoid the need (if possible) for any configuration that it just works out of the box.
For the build-time some p2-instructions could be added to the fragment, but PDE does not yet have any help for that as far as I know.